File: M3Tools/Dialogs/Listeners/AddListenerDialog.vb
Imports System.Text.RegularExpressions
Imports System.Windows.Forms
Imports SPPBC.M3Tools.Events

Namespace Dialogs
	Public Class AddListenerDialog

		Public Event ListenerAdded As Listeners.ListenerEventHandler

		Public Property ListenerName As String
			Get
				Return gi_Name.Text
			End Get
			Set(value As String)
				gi_Name.Text = value
			End Set
		End Property

		Public Property ListenerEmail As String
			Get
				Return gi_Email.Text
			End Get
			Set(value As String)
				gi_Email.Text = value
			End Set
		End Property

		ReadOnly Property Listener As Types.Listener
			Get
				Return New Types.Listener(-1, ListenerName, ListenerEmail)
			End Get
		End Property

		Private Sub AddListener(sender As Object, e As EventArgs) Handles btn_Create.Click
			If Not ValidInputs() Then
				Return
			End If

			Me.DialogResult = DialogResult.OK
			Me.Close()
		End Sub

		Private Sub Cancel(sender As Object, e As EventArgs) Handles btn_Cancel.Click
			Dim res = MessageBox.Show("Are you sure you want to cancel listener creation?", "Cancel Creation", MessageBoxButtons.YesNo, MessageBoxIcon.Question)

			If Not res = DialogResult.Yes Then
				Return
			End If

			DialogResult = DialogResult.Cancel
			Close()
		End Sub

		Private Function ValidInputs() As Boolean
			If String.IsNullOrWhiteSpace(Me.ListenerName) Then
				Return False
			End If

			If String.IsNullOrWhiteSpace(Me.ListenerEmail) OrElse Not Regex.IsMatch(Me.ListenerEmail, My.Resources.EmailRegex2) Then
				Return False
			End If

			Return True
		End Function
	End Class
End Namespace

File: M3Tools/Dialogs/Listeners/AddListenerDialog.cs
using System;
using System.Text.RegularExpressions;
using System.Windows.Forms;

namespace SPPBC.M3Tools.Dialogs
{
	/// <summary>
	/// A dialog box for adding new listeners to the ministry
	/// </summary>
	public partial class AddListenerDialog
	{
		/// <summary>
		/// The name of the new listener
		/// </summary>
		public string ListenerName
		{
			get => gi_Name.Text;
			set => gi_Name.Text = value;
		}

		/// <summary>
		/// The email of the new listener
		/// </summary>
		public string ListenerEmail
		{
			get => gi_Email.Text;
			set => gi_Email.Text = value;
		}

		/// <summary>
		/// The new listener
		/// </summary>
		public Types.Listener Listener => new(-1, ListenerName, ListenerEmail);

		/// <summary>
		/// 
		/// </summary>
		public AddListenerDialog()
		{
			InitializeComponent();
		}

		private void AddListener(object sender, EventArgs e)
		{
			if (!ValidInputs())
			{
				return;
			}

			DialogResult = DialogResult.OK;
			Close();
		}

		private void Cancel(object sender, EventArgs e)
		{
			DialogResult res = MessageBox.Show("Are you sure you want to cancel listener creation?", "Cancel Creation", MessageBoxButtons.YesNo, MessageBoxIcon.Question);

			if (!(res == DialogResult.Yes))
			{
				return;
			}

			DialogResult = DialogResult.Cancel;
			Close();
		}

		private bool ValidInputs()
		{
			if (string.IsNullOrWhiteSpace(ListenerName))
			{
				return false;
			}

			return !string.IsNullOrWhiteSpace(ListenerEmail) && Regex.IsMatch(ListenerEmail, Properties.Resources.EmailRegex2);
		}
	}
}
